pico_w/wifi/ntp_system_time/README.md

```diff
@@ -0,0 +1,92 @@
+
+The example connects to Wi-Fi and displays the local time in the UK or another timezone that you specify, synchronised once an hour to one of the servers from [pool.ntp.org](https://www.ntppool.org/en/). 
+
+Uses the SNTP application provided by lwIP and the Pico 'always-on timer' _(RTC on Pico/RP2040, powman timer on Pico-2/RP2350)_.
+
+# Running the example
+
+Provide the SSID and password of your Wi-Fi network by editing `CMakeLists.txt` or on the command line; then build and run the example as usual.
+
+You should see something like this:
+
+```
+Connecting to Wi-Fi...
+connect status: joining
+connect status: link up
+Connected
+IP address 192.168.0.100
+system time not yet initialised
+-> initialised system time from NTP
+GMT: Sun Oct 26 10:41:07 2025
+GMT: Sun Oct 26 10:41:12 2025
+...
+```
+
+### To use it in your own code
+Configure the lwIP callbacks and connect to the network as shown in the example. You can then initialise the background NTP synchronisation like this:
+
+```
+sntp_setoperatingmode(SNTP_OPMODE_POLL);
+sntp_init();
+```
+
+Your code can now call 
+
+```
+void get_time_utc(struct timespec *)
+```
+
+whenever it wants the current UTC time. 
+ 
+ You can also use the [pico_aon_timer API](https://www.raspberrypi.com/documentation/pico-sdk/high_level.html#group_pico_aon_timer) to read the time directly or to set alarms. _Note however that with a direct read it is theoretically possible (although very unlikely) to get an erroneous result if NTP was in the process of updating the timer in the background._
+
+ To reduce the logging level change the `SNTP_DEBUG` option in **lwipopts.h** to `LWIP_DBG_OFF` and/or remove the informational messages from `sntp_set_system_time_us()` in **ntp_system_time.c**.
+
+
+# Further details
+
+The example uses:
+
+1. the [SNTP application](https://www.nongnu.org/lwip/2_0_x/group__sntp.html) provided by the lwIP network stack
+2. the Pico SDK high level "always on timer" abstraction: [pico_aon_timer](https://www.raspberrypi.com/documentation/pico-sdk/high_level.html#group_pico_aon_timer)
+3. a [POSIX timezone](https://ftp.gnu.org/old-gnu/Manuals/glibc-2.2.3/html_node/libc_431.html) to convert UTC to local time
+
+### lwIP SNTP
+
+The lwIP SNTP app provides a straightforward way to obtain and process timestamps from a pool of NTP servers without the complexity of a full NTP implementation. The lwIP documentation covers the [configuration options](https://www.nongnu.org/lwip/2_0_x/group__sntp.html) but the comments in the [source code](https://github.com/lwip-tcpip/lwip/blob/master/src/apps/sntp/sntp.c) are also very helpful.
+
+SNTP uses the **macros** `SNTP_GET_SYSTEM_TIME(sec, us)` and `SNTP_SET_SYSTEM_TIME(sec, us)` to call user-provided functions for accessing the system clock. The example defines the macros in `lwipopts.h` and the callbacks themselves are near the top of `ntp_system_time.c`.
```

Comment:
Your `lwipopts.h` file declares `SNTP_GET_SYSTEM_TIME` and `SNTP_SET_SYSTEM_TIME_US` which don't quite match the macro-names you mention here?